optuna/study.py

```diff
@@ -71,8 +71,7 @@ def best_value(self):
         """
 
         best_value = self.best_trial.value
-        if best_value is None:
-            raise ValueError('No trials are completed yet.')
+        assert best_value is not None
```

Comment:
This assertion is needed for type checking by `mypy`.

Reply:
```suggestion
        assert best_value is not None, "No trials are available as this study is not used yet."
```

This change is for the below case.

```shell
In [1]: import optuna
study = opt
In [2]: study = optuna.create_study()

In [3]: study.best_trials
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-0e423b03e810> in <module>
----> 1 study.best_trials

AttributeError: 'Study' object has no attribute 'best_trials'
```

Reply:
Hmm, this assertion never fails because the above case is handled in  `BaseStorage.get_best_trial()` method (see the following backtrace):
```python
In [1]: import optuna

In [2]: study = optuna.create_study()

In [3]: study.best_value
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-226de324a2b4> in <module>
----> 1 study.best_value

/d/dev/pfn/optuna/optuna/study.py in best_value(self)
     71         """
     72
---> 73         best_value = self.best_trial.value
     74         assert best_value is not None
     75

/d/dev/pfn/optuna/optuna/study.py in best_trial(self)
     85         """
     86
---> 87         return self._storage.get_best_trial(self.study_id)
     88
     89     @property

/d/dev/pfn/optuna/optuna/storages/base.py in get_best_trial(self, study_id)
    185
    186         if len(all_trials) == 0:
--> 187             raise ValueError('No trials are completed yet.')
    188
    189         if self.get_study_direction(study_id) == structs.StudyDirection.MAXIMIZE:

ValueError: No trials are completed yet.
```

Reply:
I was confused. Yes, removing the `None` check makes sense 👍